graph/push.go

```diff
@@ -378,6 +381,12 @@ func (s *TagStore) CmdPush(job *engine.Job) engine.Status {
 	job.GetenvJson("authConfig", authConfig)
 	job.GetenvJson("metaHeaders", &metaHeaders)
 
+	// Set a header so remotes can identify the command being carried out. If
+	// present, the remote may act on the field but this is mostly advisory.
+	metaHeaders["Docker-Command"] = []string{"push"}
+
+	repoInfo.Index.Headers = metaHeaders
```

Comment:
Dirty. I really don't like it but doing it right will require (imo) the refactor we discussed, having the endpoint be the top level value from which sessions and factories are derived.